javalib/src/main/scala/java/util/BitSet.scala

```diff
@@ -0,0 +1,859 @@
+        isLengthActual = true
+      }
+    }
+
+    needClear = true
+  }
+
+  def xor(set: BitSet): Unit = {
+    val bsActualLen = set.getActualArrayLength()
+
+    if (bsActualLen > bits.length) {
+      val tempBits = new Array[Int](bsActualLen)
+      System.arraycopy(set.bits, 0, tempBits, 0, set.actualArrayLength)
+
+      for (i <- 0 until actualArrayLength)
+        tempBits(i) ^= bits(i)
+
+      bits = tempBits
+      actualArrayLength = bsActualLen
+      isLengthActual = !((actualArrayLength > 0) && (bits(actualArrayLength - 1) == 0))
+    } else {
+      val bsBits = set.bits
+
+      for (i <- 0 until bsActualLen)
+        bits(i) ^= bsBits(i)
+
+      if (bsActualLen > actualArrayLength) {
+        actualArrayLength = bsActualLen
+        isLengthActual = true
+      }
+    }
+
+    needClear = true
+  }
+
+  def andNot(set: BitSet): Unit = {
+    if (needClear) {
+      val bsBits = set.bits
+      val range =
+        if (actualArrayLength < set.actualArrayLength) actualArrayLength
+        else set.actualArrayLength
+
+      for (i <- 0 until range)
+        bits(i) &= ~bsBits(i)
+
+      if (actualArrayLength < range)
+        actualArrayLength = range
+
+      isLengthActual = !((actualArrayLength > 0) && (bits(actualArrayLength - 1) == 0))
+    }
+  }
+
+  override def hashCode(): Int = {
+    var x: Long = 1234L
+    var i: Int = 0
+
+    while (i < actualArrayLength) {
+      x ^= bits(i) * (i + 1)
+      i += 1
+    }
+
+    ((x >> 32) ^ x).toInt
+  }
+
+  def size(): Int = bits.length << BitSet.AddressBitsPerWord
+
+  override def equals(obj: Any): Boolean = {
+    // scalastyle:off return
+    if (obj.isInstanceOf[AnyRef] && (this eq obj.asInstanceOf[AnyRef]))
+      return true
+
+    if (!obj.isInstanceOf[BitSet])
+      return false
+
+    val bs: BitSet = obj.asInstanceOf[BitSet]
+    val bsBits = bs.bits
+    val length1 = actualArrayLength
+    val length2 = bs.actualArrayLength
+
+    if (isLengthActual && bs.isLengthActual && length1 != length2)
+      return false
+
+    // If one of the BitSets is larger than the other, check to see if
+    // any of its extra bits are set. If so return false.
+    if (length1 <= length2) {
+      var i: Int = 0
+      while (i < length1) {
+        if (bits(i) != bsBits(i))
+          return false
+
+        i += 1
+      }
+
+      i = length1
+      while (i < length2) {
+        if (bsBits(i) != 0)
+          return false
+
+        i += 1
+      }
+    } else {
+      var i: Int = 0
+      while (i < length2) {
+        if (bits(i) != bsBits(i))
+          return false
+
+        i += 1
+      }
+
+      i = length2
+      while (i < length1) {
+        if (bits(i) != 0)
+          return false
+
+        i += 1
+      }
+    }
+
+    true
+    // scalastyle:on return
+  }
+
+  override def clone(): AnyRef = try {
+    val clone = super.clone.asInstanceOf[BitSet]
+    clone.bits = bits.clone
+    clone
+  } catch {
+    case _: CloneNotSupportedException => null
+  }
```

Comment:
This can be simplified and be more efficient as:
```suggestion
  override def clone(): AnyRef =
    new BitSet(bits.clone(), actualArrayLength)
```